### DISCUSSION TITLE
Use getInitialProps in _app.js in a Typescript project

### DISCUSSION THREAD

**Discussion by ludersGabriel** [vercel/next.js#31183] [Help] [answered]
Hey! I'm trying to set a getInitialProps at _app.js file so I can grab information about auth and locale to pass to my Contexts before rendering any page. Problem is I'm having trouble typing it, getting the following error: 

```
Property 'getInitialProps' does not exist on type 'FC<AppProps<{}>>'
```

This is my _appjs: 
```
import { AppProps, AppContext } from 'next/app'
import { ThemeProvider } from 'styled-components'
import GlobalStyles from '../styles/global'
import theme from '../styles/theme'
import Layout from '../components/layout'

const MyApp: React.FC<AppProps> = ({ Component, pageProps }) => {
  return (
    <ThemeProvider theme={theme}>
      <Layout>
        <Component {...pageProps} />
        <GlobalStyles />
      </Layout>
    </ThemeProvider>
  )
}

MyApp.getInitialProps = async (appContext: AppContext) => {
  // calls page's `getInitialProps` and fills `appProps.pageProps`

  console.log('bruh')

  return {}
}

export default MyApp
```

I've tried changing React.FC for NextPage, but the props on _appjs are not supported by NextPage. What should I do?

**Comment by tilap** [comment, marked as answer]
As you have `getInitialProps`, you cannot use `React.FC` (read https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L546)

Try:

```
const MyApp = ({ Component, pageProps }: AppProps) => {
   // ...
}

MyApp.getInitialProps = async (ctx: AppContext) => {
  // ...
}

export default MyApp;
```

**Reply by ludersGabriel** [reply]
That does resolve the get initial props, but I'm still unable to build the project because now the props have no type:

![image](https://user-images.githubusercontent.com/57847215/140915482-071eb12b-e8e2-4b4d-8cd2-7a4fd5cc92a7.png)

Any tips?

**Reply by tilap** [reply]
`MyApp` props have no type in your example.
Watch my code example closer: `({ Component, pageProps }: AppProps)`

**Reply by ludersGabriel** [reply]
Yooo, my bad! It worked like a charm, thanks dude, you're awesome
